### PR TITLE
add more branch protection checks in checkov

### PR DESCRIPTION
**Pull request by schosterbarak**
add more github repository rules around branch protectioni
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: tests/github/resources/github_conf/repo/branch_protection.json [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+          "site_admin": false
+        }
+      ],
+      "teams": [
+        {
+          "id": 1,
+          "node_id": "MDQ6VGVhbTE=",
+          "url": "https://api.github.com/teams/1",
+          "html_url": "https://github.com/orgs/github/teams/justice-league",
+          "name": "Justice League",
```
Comment (gruebel) [inline tests/github/resources/github_conf/repo/branch_protection.json:49]: Avengers are better 😄

File: checkov/github/dal.py [hunk 2]
```diff
@@ -60,10 +60,17 @@ def _authenticate(self):
...
     def _request(self, endpoint):
+        if not self.token:
+            return
         url_endpoint = "{}{}".format(self.api_url, endpoint)
         try:
-            return self.http.request("GET", url_endpoint,
-                                     headers=self._headers())
+            request = self.http.request("GET", url_endpoint,
+                                        headers=self._headers())
+            if request.status == 200:
```
Comment (gruebel) [inline checkov/github/dal.py:69]: shouldn't you also catch the non `200` status codes?

Comment (schosterbarak) [reply checkov/github/dal.py:69]: failing silently

**Review by gruebel — APPROVED** [review]
🏖️

**Pull request merged**